src/videojs-hls.js

```diff
@@ -474,13 +474,20 @@ videojs.Hls.prototype.fillBuffer = function(offset) {
   }
 
   // resolve the segment URL relative to the playlist
+  segmentUri = this.resolveSegmentUrl(segment.uri);
+
+  this.loadSegment(segmentUri, offset);
+};
+
+videojs.Hls.prototype.resolveSegmentUrl = function(segmentRelativeUrl) {
```

Comment:
I'm wondering if there's a better name for this method now that we'll be using it for more than just segment URL resolution.